### PR TITLE
Add to cart

### PR DESCRIPTION
**Pull request by korkoza**
1. added cartPage, itemPage objectpages
2. didn't manage to create and use checkboxElement objectpage as different elements are used for selection and verification. 
3. Changed some css selectors for pageBaseElement

### CODE REVIEW

**Comment by korkoza** [comment]
1. Added selectCheckbox(), unSelectCheckbox() methods.
2. refactored 06_add_to_cart using created methods

Please review

File: pages/MainPage.js [hunk 1]
```diff
@@ -10,7 +10,8 @@ class MainPage extends BasePage {
     }
 
     getPageBaseElement() {
-        return new BaseElement(element(by.css(".text-center")), "Main banner");
+        return new BaseElement(element(by
+            .css(".thumbnails.owl-loaded.owl-drag > div.owl-stage-outer > div")), "Main banner");
```
Comment (nbabii) [inline pages/MainPage.js]: Try to use shorter locator

File: pages/MainPage.js [hunk 2]
```diff
@@ -49,7 +50,29 @@ class MainPage extends BasePage {
...
+    getTreeElement() {
+        return new BaseElement(element(by.css('a[href="/dacha_sad/"]')), 'Tree element');
+    }
+
+    getSubTreeElement() {
+        return new BaseElement(element(by
+            .css('li[data-menu-id="2952"]')), 'Subtree element');
+    }
+
+    getCategoryElement() {
```
Comment (nbabii) [inline pages/MainPage.js]: you probably need to describe exact category in method name

File: pages/MainPage.js [hunk 3]
```diff
@@ -49,7 +50,29 @@ class MainPage extends BasePage {
...
+        return new BaseElement(element(by
+            .css('a[href*="/dacha_sad/nasosy-vodosnabzheniya/46036/"]')), 'Category element');
+    }
+
+    async navigateCategory() {
```
Comment (nbabii) [inline pages/MainPage.js]: you probably need to describe exact category in method name

File: pages/MainPage.js [hunk 4]
```diff
@@ -49,7 +50,29 @@ class MainPage extends BasePage {
             await this.getDoSearchElement().click();
         })();
     }
-}
 
+    getTreeElement() {
```
Comment (nbabii) [inline pages/MainPage.js]: you probably need to describe exact name in method name

File: pages/cartPage.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        await this.getPageBaseElement().waitForVisible();
+    }
+
+    getPageBaseElement() {
+        return new BaseElement(element(by.css(".cart-info.text")), "Product's list");
+    }
+
+    // Delivery type dropdown elements
+
+    getDeliveryTypeDropdown() {
```
Comment (nbabii) [inline pages/cartPage.js]: You need to create new element type for drop down and use it here (like we did with checkbox). This drop-down element should have method to select option by text (provided as attribute). So method should click on drop down to open it and select option by text.

File: pages/cartPage.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        return new BaseElement(element(by.css("div.viewbox:nth-child(2) .field.m_b-sm")), "Delivery dropdown");
+    }
+
+    getDeliveryTypeOption() {
```
Comment (nbabii) [inline pages/cartPage.js]: check comment above

File: pages/cartPage.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        return new BaseElement(element(by.css("option[data-delivery-id='13261']")), "Delivery type #1");
+    }
+
+    async setDeliveryType() {
+        await allure.createStep("Select delivery type", async () => {
+            await this.getDeliveryTypeDropdown().click();
```
Comment (nbabii) [inline pages/cartPage.js]: check comment above, so here this drop down element should be used

File: pages/cartPage.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+            await this.getDeliveryTypeOption().click();
+        })();
+    }
+
+    // Department dropdown elements
+    getDepartmentDropdown() {
```
Comment (nbabii) [inline pages/cartPage.js]: check comment above

File: pages/cartPage.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        return new BaseElement(element(by.css("div[data-delivery-type-options='warehouseNP']")), "Department dropdown");
+    }
+
+    getDepartmentOption() {
```
Comment (nbabii) [inline pages/cartPage.js]: check comment above

File: pages/cartPage.js [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+        return new BaseElement(element(by
+            .css("div[data-delivery-type-options='warehouseNP'] option:nth-child(2)")), "Department option");
+    }
+
+    async setDepartmentOption() {
```
Comment (nbabii) [inline pages/cartPage.js]: check comment above

File: pages/MainPage.js (hunk 1 above)
Comment (korkoza) [reply pages/MainPage.js]: changed to "".thumbnails > .owl-stage-outer""

File: pages/MainPage.js (hunk 2 above)
Comment (korkoza) [reply pages/MainPage.js]: renamed to getNasosCategoryElement()

File: pages/MainPage.js (hunk 4 above)
Comment (korkoza) [reply pages/MainPage.js]: renamed to getDachaSadTreeElement()
also renamed getSubreeElement() to getBaseinSubTreeElement()

File: pages/cartPage.js (hunk 5 above)
Comment (korkoza) [reply pages/cartPage.js]: fixed

File: pages/cartPage.js (hunk 6 above)
Comment (korkoza) [reply pages/cartPage.js]: fixed

File: pages/cartPage.js (hunk 7 above)
Comment (korkoza) [reply pages/cartPage.js]: fixed

File: pages/cartPage.js (hunk 8 above)
Comment (korkoza) [reply pages/cartPage.js]: fixed

File: pages/cartPage.js (hunk 9 above)
Comment (korkoza) [reply pages/cartPage.js]: fixed

File: pages/cartPage.js (hunk 10 above)
Comment (korkoza) [reply pages/cartPage.js]: fixed

**Comment by korkoza** [comment]
also added attribute to getCartIcon() method

File: elements/BaseElement.js [hunk 11]
```diff
@@ -15,9 +15,23 @@ class BaseElement {
...
+    async getAttribute(attribute) {
+        console.log(`Get attribute "${attribute}" of "${this.name}"`);
+        return await this.protractorElement.getAttribute(attribute);
+    }
+
     async waitForVisible() {
         await browser.wait(EC.visibilityOf(this.getProtractorElement()), browser.params.explicitWait);
     }
+
+    async elementToBeSelected() {
```
Comment (nbabii) [inline elements/BaseElement.js]: Please rename to `waitForElementToBeSelected`

File: elements/dropdown.js [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+
+let optionLocator = "option";
+
+
+class DropdownElement extends BaseElement {
+    async selectOption(optionText) {
+        let innerOption = new BaseElement(this.getProtractorElement()
+            .element(by.cssContainingText(optionLocator, optionText)), "Option by text");
+        await this.getProtractorElement().click();
+        await innerOption.click();
```
Comment (nbabii) [inline elements/dropdown.js]: Good work, just do in this way (open and after find and click):
```
await this.getProtractorElement().click();
let innerOption = new BaseElement(this.getProtractorElement()
            .element(by.cssContainingText(optionLocator, optionText)), "Option by text");        
await innerOption.click();
```

File: pages/MainPage.js [hunk 13]
```diff
@@ -49,7 +50,29 @@ class MainPage extends BasePage {
             await this.getDoSearchElement().click();
         })();
     }
-}
 
+    getDachaSadTreeElement() {
```
Comment (nbabii) [inline pages/MainPage.js]: It is better to use English words =), but let it be this time.

File: elements/BaseElement.js (hunk 11 above)
Comment (korkoza) [reply elements/BaseElement.js]: fixed

File: elements/dropdown.js (hunk 12 above)
Comment (korkoza) [reply elements/dropdown.js]: fixed

File: pages/MainPage.js (hunk 13 above)
Comment (korkoza) [reply pages/MainPage.js]: I was hesitating regarding this :)

**Pull request merged**